flutterproject/flutterproject.go

```diff
@@ -0,0 +1,50 @@
+package flutterproject
+
+type SDKVersionsReader interface {
+	ReadSDKVersions(projectRootDir string) (*VersionConstraint, *VersionConstraint, error)
+}
+
+type Project struct {
+	rootDir    string
+	fileOpener FileOpener
+}
+
+func New(rootDir string, fileOpener FileOpener) Project {
+	return Project{
+		rootDir:    rootDir,
+		fileOpener: fileOpener,
+	}
+}
+
+type FlutterAndDartSDKVersions struct {
+	FlutterSDKVersions []VersionConstraint
+	DartSDKVersions    []VersionConstraint
+}
+
+func (p Project) FlutterAndDartSDKVersions() (FlutterAndDartSDKVersions, error) {
+	versionReaders := []SDKVersionsReader{
+		NewFVMVersionReader(p.fileOpener),
+		NewASDFVersionReader(p.fileOpener),
+		NewPubspecLockVersionReader(p.fileOpener),
+		NewPubspecVersionReader(p.fileOpener),
+	}
```

Comment:
I am not familiar with Flutter so the question I will be writing might be wrong. 

We have 4 readers and the loop iterates over all of them and collects the results of all 4 and actually returns the results for all 4 as it uses arrays. Is is a possibility to have multiple flutter and dart version inside a project. My first thought would be that a project has only a single one. So my question is that can we deduplicate the versions in this function and only return single ones instead of an array?

Reply:
I plan to use this function for 2 things:
- tracking how Flutter projects are specifying Flutter SDK requirements (are they using a third-party version manager [fvm or asdf]? -> how should we manage Flutter SDk versions on the stacks) and what are the typical requirements (what SDK versions should we preinstall on stacks)
- calculate what Flutter SDK version should be installed based on the project and make the project scanner configure the Flutter Installer based on that. I plan to create a new function that returns a single SDK version to support the scanner.